### ISSUE TITLE
scroll-marker from dom?

### ISSUE THREAD

**Issue by bkardell** [flackr/carousel#6]
If I am understanding properly, the really powerful thing about the [`scroll-marker`](https://flackr.github.io/carousel/examples/scroll-marker/) idea is that it lets you write something to style the 'active one' with `:checked`, like `::scroll-marker:checked { background: pink; }`.  This would be super to be able to mix with the [`grid-flow` example](https://flackr.github.io/carousel/examples/grid-flow/tabs/#details) because very often it would be the case where if using this for something more like 'tabs' (panelset-style tabs at least) it's not likely to be possible or easy to do that part in CSS.  Is there a way you could do this?  The only one I could maybe think of is via `:attr()`, but ideally that would not be just through an attribute for internationalization/localization reasons

**Comment by flackr** [comment]
Right, so currently in the grid-flow example they're not scroll markers. If you wanted to make the non-pseudo-element tab headers be scroll markers then we have to explain how establish the link between the scroll-marker element and the content it wants to scroll to, e.g.

```html
<scroll-marker for="main">Main</scroll-marker>
<section id="main">...</section>
```

Element scroll markers is something that I want to have eventually, however ideally that linkage between marker and content to scroll to would be established without names, e.g. maybe using a named css property, maybe it's the immediate ancestor? It gets more complicated so I figured starting with pseudo-element only markers was a good starting place. I forgot to mention this in the explainer, I'll add that.

If you wanted to make them as pseudo-element scroll-markers then you could put the title on the content element, and do something like this (which is I think what you were suggesting above:
```html
<style>
section::scroll-marker {
  content: attr(name);
}
</style>
<section name="Main">...</section>
<section name="Details">...</section>
<section name="Followup">...</section>
```

**Comment by flackr** [comment]
Of course another way you could style the active marker differently in that example is by leveraging the `:target` pseudo-class to select the link preceding the current target section:
```css
a:has(+ :target) {
  /* Active style */
}
```

This won't work in the polyfill right now because it has to restructure the DOM in order to fake the properties it's trying to polyfill but would work when implemented for real.

**Comment by bkardell** [comment]
Yeah, I was thinking both of those things too.... It might work if we integrated it into https://tabvengers.github.io/spicy-sections/ though 🤔 right? I mean, even a simple version of that that didn't switch or support other affordances but just 'projected' them

**Comment by flackr** [comment]
I've been thinking a bit more about this problem as we would like to allow developer DOM elements to function as scroll markers. Also it has been suggested by the invokers proposal (see #2) and the carousel discussion https://github.com/w3c/csswg-drafts/issues/9745#issuecomment-1944824869 that using ID's as our initial support for targeting may be good enough for some common cases. I think it would be good to flesh this out assuming that for now and we can always add something like `timeline-scope` name lifting if we need more dynamic CSS targeting in the future.

Assuming ID targeting for now and that we have some element with ID `#target`, I think there are a few high level ways we could make this happen:
1. A new element with an attribute giving the target. E.g.
   ```html
   <scrollmarker target="#foo">Foo</scrollmarker>
   ```
2. A property on an existing element, most likely on the `<button>` element as it is similar to `popovertarget`. E.g.
   ```html
   <button type="scroll" target="#foo">Foo</button>
   ```
3. A CSS property. E.g.
   ```html
   <style>
     .marker {
       scroll-marker-target: #foo;
     }
   </style>
   <a class="marker" href="#foo">Foo</a>
   ```

Option 3 is included for completeness, but I do not think it is a good idea to have a CSS property change the behavior of an element. Options 1 and 2 both seem workable. My preference would be to go for option 2 given that I think it provides a better progressive enhancement story (the button will still render, but just needs script handlers) and feels similar to other proposals (e.g. invokers, popover, etc).

There are a few behaviors of scroll markers which are unusual for buttons (but fairly normal for radio buttons):
* They should scroll to the target on focus.
* Only one of the scroll markers for a given scroller should be active (and focusable) at a time, arrow keys should cycle between the others.
* They are automatically selected as a result of scrolling.

Perhaps given that they have a new type the change in behavior would be reasonable, but we could also consider whether one or more of these behaviors should be keyed on some other property.

**Comment by flackr** [comment]
I've added two competing proposals for how this could be done, one based on extending invokers / focusgroup and the other based on a new proposal similar to invokers but specifically targeting the requirements of scroll markers.

Adding @mfreed7 as the invokers direction was discussed in the context of #2.

**Comment by mfreed7** [comment]
Looks like an interesting cross of invokers and `focusgroup`! + @travisleithead for reference.
